packages/box/src/Box.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { HTMLElementProps } from '@leafygreen-ui/lib';
+import omit from 'lodash/omit';
+
+type DivProps<T> = HTMLElementProps<'div'> & T;
+
+type AnchorProps<T> = HTMLElementProps<'a'> &
+  T & {
+    href: string;
+  };
+
+type CustomElementProps<T> = T & {
+  component: React.ElementType<any>;
+  [key: string]: any;
+};
+
+export type BoxProps<T> = DivProps<T> | AnchorProps<T> | CustomElementProps<T>;
+
+function usesCustomElement<T>(
```

Comment:
```suggestion
function isCustomElement<T>(
```

And ditto `isAnchorElement`?

WDYT?

Reply:
Works for me